paasta_tools/chronos_tools.py

```diff
@@ -225,7 +226,17 @@ def get_env(self):
         return [{"name": key, "value": value} for key, value in original_env.iteritems()]
 
     def get_constraints(self):
-        return self.config_dict.get('constraints', [['pool', 'EQUALS', self.get_pool()]])
+        if 'constraints' in self.config_dict:
+            constraints = self.config_dict.get('constraints')
+        else:
+            constraints = self.config_dict.get('extra_constraints', [])
+            constraints.extend(self.get_deploy_constraints())
+            constraints.extend(self.get_pool_constraints())
+        assert isinstance(constraints, list)
```

Comment:
Welll... 

I forgot which particular bug I was traking tracking down, but this helped get it.

Do we have a policy with sprinkling asserts around where we *know* what better be returned by a function?